tests/phpunit/tests/blocks/registerCoreBlockStyleHandles.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * Tests for block style handles.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ *
+ * @since 6.3.0
+ *
+ * @group blocks
+ *
+ * @covers ::register_core_block_style_handles
+ */
+class Tests_Blocks_registerCoreBlockStyleHandles extends WP_UnitTestCase {
+	private $old_wp_styles;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->old_wp_styles = $GLOBALS['wp_styles'];
+
+		remove_action( 'wp_default_styles', 'wp_default_styles' );
+
+		if ( empty( $GLOBALS['wp_styles'] ) ) {
+			$GLOBALS['wp_styles'] = null;
+		}
+	}
+
+	public function tear_down() {
+		$GLOBALS['wp_styles'] = $this->old_wp_styles;
+
+		add_action( 'wp_default_styles', 'wp_default_styles' );
+
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 58528
+	 */
+	public function test_wp_should_load_separate_core_block_assets_false() {
+		register_core_block_style_handles();
+
+		$style_fields = array(
+			'style'       => 'style',
+			'editorStyle' => 'editor',
+		);
+
+		$core_blocks_meta = $this->get_block_data();
+
+		foreach ( $core_blocks_meta as $name => $schema ) {
+
+			if ( ! isset( $schema['style'] ) ) {
+				$schema['style'] = "wp-block-$name";
+			}
+			if ( ! isset( $schema['editorStyle'] ) ) {
+				$schema['editorStyle'] = "wp-block-{$name}-editor";
+			}
+			foreach ( $style_fields as $style_field => $filename ) {
+				$style_handle = $schema[ $style_field ];
+				if ( is_array( $style_handle ) ) {
+					continue;
+				}
+
+				$this->assertArrayNotHasKey( $style_handle, $GLOBALS['wp_styles']->registered, 'The key should not exist, as this style should not be registered' );
+			}
+		}
+	}
+
+
+	/**
+	 * @ticket 58528
+	 */
+	public function test_wp_should_load_separate_core_block_assets_true() {
+		add_filter( 'should_load_separate_core_block_assets', '__return_true' );
+		register_core_block_style_handles();
+
+		$style_fields = array(
+			'style'       => 'style',
+			'editorStyle' => 'editor',
+		);
+
+		$core_blocks_meta = $this->get_block_data();
+		$wp_styles        = $GLOBALS['wp_styles'];
+		$includes_url     = includes_url();
+
+		foreach ( $core_blocks_meta as $name => $schema ) {
+
+			if ( ! isset( $schema['style'] ) ) {
+				$schema['style'] = "wp-block-$name";
+			}
+			if ( ! isset( $schema['editorStyle'] ) ) {
+				$schema['editorStyle'] = "wp-block-{$name}-editor";
+			}
```

Comment:
Another benefit of using a data provider is that you'll be able to do this in the data provider method rather than duplicated across both tests.